### ISSUE TITLE
Deleting disease from disease modal - check for Provisional, provide warning note if Provisional

### ISSUE THREAD

**Issue by selinad** [ClinGen/clincoded#1010]
Currently, a curator could mark a disease-associated interpretation with a calculated or modified pathogenicity of LP or P Interpretation as provisional and then go and remove the disease from the interpretation using the disease modal.

For this release, if a curator deletes a disease in the modal, have it check whether Provisional.  If provisional, display warning note before hitting "OK:"

Warning: This interpretation is marked as "Provisional." If it has a modified pathogenicity of "Likely pathogenic" or "Pathogenic," or no modified pathogenicity but a Calculated Pathogenicity of "Likely pathogenic" or "Pathogenic," it must be associated with a disease.  If you still wish to delete the disease, first go to the Evaluation Summary page and remove the "Provisional" tag.

(this could go in yellow box) - edits welcome!

Note - more logic will be added to take care of this outlier case in the next release

**Comment by jimmyzhen** [comment]
Hi @selinad, @wrightmw,

I have implemented the requested changes in this ticket. Below is a screenshot of what I have done so far:
![image](https://cloud.githubusercontent.com/assets/6956310/18882105/bc1c38cc-8492-11e6-95ed-713fbd06c1fb.png)

I will spin up an instance for you to review.

Due to its dependency on the provisional interpretation checkbox value, I have merged my branch for the summary page (#943) into this branch. As a result, the instance that I am about to spin up will include all Summary page changes as well as disease modal changes described in this ticket.

**Comment by jimmyzhen** [comment]
Hi @selinad, @wrightmw,

You may review the changes in ticket via the following instance:
https://1010-jz-disease-modal-2c85e97-jzhen.demo.clinicalgenome.org

**Comment by selinad** [comment]
Hi @jimmyzhen things are looking great - 

One thing - if I do ignore message for a Provisional with an LP and delete the disease, then go to Summary, it shows this:

![screen shot 2016-09-27 at 12 15 09 pm](https://cloud.githubusercontent.com/assets/11320314/18888204/2fe2e472-84ac-11e6-9e2b-520dd7f11068.png)

This may be fine for now, but should it reset the modify and reason just like it resets Mark as provisional?

**Comment by wrightmw** [comment]
@selinad I think you are right. The modify and reason should probably be reset if the disease is deleted. Likewise, maybe they should also be reset if the disease is changed.

**Comment by selinad** [comment]
@jimmyzhen It seems everything is behaving as expected with exceptions of above comments. Nicely done! It will be good to see it integrated with @kgliu0101 instance/work on static bar.

One change to wording:

Warning: This interpretation is marked as "Provisional." If it has a Modified Pathogenicity of "Likely pathogenic" or "Pathogenic," or no Modified Pathogenicity but a Calculated Pathogenicity of "Likely pathogenic" or "Pathogenic," it must be associated with a disease. 

**If you still wish to delete the disease, select "Cancel," then select "View Summary" and remove the "Provisional" selection** - otherwise, deleting the disease will automatically remove the "Provisional" status.

@wrightmw please feel free to edit!

**Comment by wrightmw** [comment]
@selinad I'm happy with that wording

**Comment by wrightmw** [comment]
I agree with @selinad it would be really useful to test an instance that integrates both @jimmyzhen 's work on the Summary page and @kgliu0101 's work on the green bar.

**Comment by jimmyzhen** [comment]
Hi @selinad, @wrightmw,

1. The warning message has been updated in my branch for this ticket.
2. In discussing with Selina in regards to a change in associated disease would reset the modified patho/reason, we have determined that there would be other use cases in which retaining the modified patho/reason values is desirable.
3. In the process, I discovered a bug in which saving any change (e.g. add/delete) in the associated disease would remove any pre-existing provisional checkbox value. The cause appears to be the missing "markAsProvisional" property in the flatten interpretation object when saving the disease causes an update in the interpretation. I have addressed this bug in my branch, as well as to uncheck the provisional checkbox if it was previously marked as provisional (upon deleting disease). However, it still needs more work because the checkbox UI change has not yet been wired up to invoke a database update.

In the interest of time, I am committing what I have at this point so that @kilodalton can merge everyone's work and build out a combined instance for you to review. And I will continue to debug in parallel.

**Comment by selinad** [comment]
Sounds great @jimmyzhen - thank you.  And, thank you for being on top of 2 above - I was confused, but you had it right!

I think it will be good to test a merged version....thank you!

**Comment by jimmyzhen** [comment]
I am now able to uncheck the provisional checkbox (and disable it) in the UI and reset the provisional to "In Progress" in the database simultaneously, in the event in which the user deletes the associated disease from the interpretation while the calculated/modified pathogenicity are either **Likely Pathogenic** or **Pathogenic**.

The changes will be merged into the combined version for review.

**Comment by selinad** [comment]
Hi @jimmyzhen this looks great!  Nice job on very tough logic.  The only problem I am having is that deleting a disease w/Provisional tag  for a LP or P gives the right behavior on the Summary page (remove Provisional), but not in green bar - even after more than a minute it still says Provisional - goes to In progress if I go to Summary page and come back...

**Comment by wrightmw** [comment]
@jimmyzhen I am seeing the same issue highlighted by @selinad above.

@selinad Comment for a future release. If a curator has evaluated disease-specific criteria and then changes the disease, should we provide a friendly note that says "If you have made any evaluations for disease-specific criteria, please check them to make sure they still apply to the new disease you have selected"

**Comment by kilodalton** [comment]
Included in the last release (R7). This included the Variant Curation Interface. Nice job and thanks for your hard work.
